widget/table.go

```diff
@@ -15,50 +15,41 @@ const tableDividerThickness = 1
 // Declare conformity with Widget interface.
 var _ fyne.Widget = (*Table)(nil)
 
+// CellID is a type that represents a cell's position in a table based on it's row and column location.
+type CellID struct {
+	Row int
+	Col int
+}
+
 // Table widget is a grid of items that can be scrolled and a cell selected.
 // It's performance is provided by caching cell templates created with CreateCell and re-using them with UpdateCell.
 // The size of the content rows/columns is returned by the Length callback.
 type Table struct {
 	BaseWidget
 
-	Length     func() (int, int)
-	CreateCell func() fyne.CanvasObject
-	UpdateCell func(row, col int, template fyne.CanvasObject)
-	// OnSelectionChanged is called whenever the selection changes, such as through user tap.
-	// The row and col values will be -1 if no cell is currently selected.
-	OnSelectionChanged func(row, col int)
-
-	selectedRow, selectedColumn int
-	hoveredRow, hoveredColumn   int
-	cells                       *tableCells
-	moveCallback                func()
-	offset                      fyne.Position
-	scroll                      *ScrollContainer
+	Length       func() (int, int)
+	CreateCell   func() fyne.CanvasObject
+	UpdateCell   func(id *CellID, template fyne.CanvasObject)
```

Comment:
ID here is just a value param and cannot be nil, so don't use pointers

Reply:
Resolved